src/number_theory/dioph.lean

```diff
@@ -109,8 +109,11 @@ namespace vector3
 @[pattern] def cons {α} {n} (a : α) (v : vector3 α n) : vector3 α (succ n) :=
 λi, by {refine i.cases' _ _, exact a, exact v}
 
+/- We do not want to make the following notation global, because then these expressions will be
+overloaded, and only the expected type will be able to disambiguate the meaning. Worse: Lean will
+try to insert a coercion from `vector3 α _` to `list α`, if a list is expected. -/
+localized "notation `[` l:(foldr `, ` (h t, vector3.cons h t) nil `]`) := l" in vector3
```

Comment:
As discussed on Zulip, it might be better to not make the notation `localized` but merely `local`. (https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/remove.20all.20instances.20with.20variable.20domain/near/177088768)

Reply:
This is a different issue -- that has to do with coercions. It doesn't matter if lemmas are stated with or without notation. There could be a problem if we put more in the `vector3` locale, and force people to get this bad notation along with other things they may want. But if this is the only thing in the locale it should be fine.